### PR TITLE
feat: island sub-directories

### LINKED ISSUE #1004: feature request: sub-folder support for islands
I've a growing number of islands.  It would be great to organise them in sub-folders, but this is not currently allowed.

### PR DESCRIPTION
**Pull request by iuioiua**
This change adds support for sub-directories in the `/island` folder and simplifies the path collection logic for routes and islands.

Closes #1004.
Supercedes #1030.

### CODE REVIEW

**Comment by iuioiua** [comment]
I suspect the current CI errors on Windows relate to paths. Any help from someone with a Windows machine is welcome.

File: src/dev/mod.ts [hunk 1]
```diff
@@ -28,70 +28,31 @@ export function ensureMinDenoVersion() {
...
 
+async function collectDir(dir: string): Promise<string[]> {
+  const paths = [];
+  const routesFolder = walk(dir, {
+    includeDirs: false,
+    includeFiles: true,
+    exts: ["tsx", "jsx", "ts", "js"],
+  });
+  for await (const entry of routesFolder) {
+    const path = SEP + relative(dir, entry.path);
```
Comment (ayame113) [inline src/dev/mod.ts]: I'm a passing Windows machine user. If you change it as follows, it should work on Windows!

```suggestion
async function collectDir(dir: string): Promise<string[]> {
  const dirUrl = toFileUrl(dir);
  const paths = [];
  const routesFolder = walk(dir, {
    includeDirs: false,
    includeFiles: true,
    exts: ["tsx", "jsx", "ts", "js"],
  });
  for await (const entry of routesFolder) {
    const path = toFileUrl(entry.path).href.substring(
      dirUrl.href.length,
    );
```

(need to add `toFileUrl` to the import statement)

**Review by lino-levan — APPROVED** [review]
LGTM. Really nice cleanup + nice change in functionality.

I was going to request an update for the docs but I realize we never had docs saying you couldn't have folders in islands 😬.

**Comment by iuioiua** [comment]
PTAL, @lucacasonato

**Review by lucacasonato — APPROVED** [review]
LGTM, nice cleanup.

**Pull request merged**